### PR TITLE
"taint" logic improvements and other minor fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/core/main.c [hunk 1]
```diff
@@ -2394,6 +2394,14 @@ int main(int argc, char *argv[]) {
                  "Loaded units and determined initial transaction in %s.",
                  format_timespan(timespan, sizeof(timespan), after_startup - before_startup, 100 * USEC_PER_MSEC));
 
+        if (arg_system) {
+                _cleanup_free_ char *taint;
+
+                taint = manager_taint_string(m);
+                if (!isempty(taint))
+                        log_notice("System is tainted: %s", taint);
```
Comment (keszybz) [inline src/core/main.c:2403]: I'd love to see a catalog entry for this. But that can and should be a separate patch.

File: src/core/manager.c [hunk 2]
```diff
@@ -3889,6 +3891,14 @@ char *manager_taint_string(Manager *m) {
         if (!streq(NOBODY_GROUP_NAME, "nobody"))
                 e = stpcpy(e, "weird-nobody-group:");
 
+        r = read_one_line_file("/proc/sys/kernel/overflowuid", &overflowuid);
+        if (r >= 0 && !streq(overflowuid, "65534"))
+                e = stpcpy(e, "overflowuid-not-65534");
```
Comment (keszybz) [inline src/core/manager.c]: Missing ":".

File: src/core/manager.c [hunk 3]
```diff
@@ -3889,6 +3891,14 @@ char *manager_taint_string(Manager *m) {
...
+
+        r = read_one_line_file("/proc/sys/kernel/overflowgid", &overflowgid);
+        if (r >= 0 && !streq(overflowgid, "65534"))
+                e = stpcpy(e, "overflowgid-not-65534");
```
Comment (keszybz) [inline src/core/manager.c]: Ditto.

Comment (poettering) [reply src/core/manager.c]: ouch!

File: src/core/manager.c (hunk 2 above)
Comment (poettering) [reply src/core/manager.c]: ouch!

**Comment by poettering** [comment]
force pushed a new version. PTAL

**Comment by keszybz** [comment]
```
Dec 07 16:38:26 autopkgtest systemd[1]: Failed to start Network Manager Wait Online.
FAIL
```
I guess that's unrelated.

**Comment by mbiebl** [comment]
There is no justification why "nogroup" is a bad idea. @poettering can you please eloborate.
So far no proper reason was given why we should taint all Debian/Ubuntu based systems by default. That makes no sense.
